packages/apollo-engine-reporting/src/agent.ts

```diff
@@ -325,12 +396,41 @@ export class EngineReportingAgent<TContext = any> {
       });
     }
 
+    if (this.options.endpointUrl) {
+      this.logger.warn(
+        'The endpointUrl option for engine has been deprecated. Please use tracesEndpointUrl instead',
+      );
+    }
```

Comment:
Nice! Let's just put `[deprecated]` as the prefix on this to be consistent with other deprecation warning patterns in this monorepo.  (This is more like a general consistency principle that this repo aims for.)
```suggestion
    if (this.options.endpointUrl) {
      this.logger.warn(
        '[deprecated] The `endpointUrl` option within `engine` has been renamed to `tracesEndpointUrl`.',
      );
    }
```